packages/fast-usdc/src/exos/liquidity-pool.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * @import {Zone} from '@agoric/zone';
+ * @import {TypedPattern} from '@agoric/internal'
+ * @import {StorageNode} from '@agoric/internal/src/lib-chainStorage.js'
+ * @import {MakeRecorderKit} from '@agoric/zoe/src/contractSupport/recorder.js'
+ * @import {USDCProposalShapes, ShareWorth} from '../pool-share-math.js'
+ */
+
+import { AmountMath } from '@agoric/ertp/src/amountMath.js';
+import {
+  AmountShape,
+  PaymentShape,
+  RatioShape,
+} from '@agoric/ertp/src/typeGuards.js';
+import { depositToSeat } from '@agoric/zoe/src/contractSupport/zoeHelpers.js';
+import { SeatShape } from '@agoric/zoe/src/typeGuards.js';
+import { M } from '@endo/patterns';
+import {
+  makeRecorderTopic,
+  TopicsRecordShape,
+} from '@agoric/zoe/src/contractSupport/topics.js';
+import {
+  deposit as depositCalc,
+  makeParity,
+  withdraw as withdrawCalc,
+  withFees,
+} from '../pool-share-math.js';
+import { makeProposalShapes } from '../type-guards.js';
+
+/**
+ * @param {Zone} zone
+ * @param {ZCF} zcf
+ * @param {Record<'USDC', Brand<'nat'>>} brands
+ * @param {{
+ *   makeRecorderKit: MakeRecorderKit;
+ * }} tools
+ */
+export const prepareLiquidityPoolKit = (zone, zcf, brands, tools) => {
+  return zone.exoClassKit(
+    'Fast Liquidity Pool',
+    {
+      feeSink: M.interface('feeSink', {
+        receive: M.call(AmountShape, PaymentShape).returns(M.promise()),
+      }),
+      external: M.interface('external', {
+        setShareWorth: M.call(RatioShape).returns(M.promise()),
+      }),
+      depositHandler: M.interface('depositHandler', {
+        handle: M.call(SeatShape, M.any()).returns(M.promise()),
+      }),
+      withdrawHandler: M.interface('withdrawHandler', {
+        handle: M.call(SeatShape, M.any()).returns(M.promise()),
+      }),
+      public: M.interface('public', {
+        makeDepositInvitation: M.call().returns(M.promise()),
+        makeWithdrawInvitation: M.call().returns(M.promise()),
+        getPublicTopics: M.call().returns(TopicsRecordShape),
+      }),
+    },
+    /**
+     * @param {ZCFMint<'nat'>} shareMint
+     * @param {StorageNode} node
+     */
+    (shareMint, node) => {
```

Comment:
There's really only one of these in the vat so no need to keep these arguments or much of the state constructor in state. You could move these up to the `prepare` call as sort of "static" property of the LiquidityPool class. `proposalShapes`, `shareWorthRecorderKit` and others could come up too to the closure.

Reply:
The `shareMint` isn't available in the 1st crank when all the kinds have to be defined. It comes from a remote call to the zoe vat.

Maybe I'm confused about the constraints in this area? I asked around internally before coming up with this solution but didn't manage to engage much with anybody. Happy to talk it over.

Reply:
Ok, I think I get it now.
- We want a ZCF Mint so the minting and burning can be synchronous.
- Making a ZCF Mint still requires a remote call, so it can't happen until after all the kinds are defined.
- Therefore the mint can't be a param to the LiquidityPool kind.

But since the pool exo is a singleton, we could define it after the remote call with a `zone.exo`. It only really needs one external facet; the rest are helpers which needn't be facets if this is a single instance. 

Then you wouldn't have to duplicate in `state` the values that are idempotently derived from the issuer.

I'd like to see this explored before I approve